encoding/av1-roi-calculator/js/av1-roi-calculator.js

```diff
@@ -0,0 +1,156 @@
+$(function() {
+
+  /**
+   * https://stackoverflow.com/questions/17369098/simplest-way-of-getting-the-number-of-decimals-in-a-number-in-javascript
+   * @param n {number}
+   * @returns {number}
+   */
+  function countDecimals(n) {
+    if (Math.floor(n.valueOf()) === n.valueOf()) {
+      return 0;
+    }
+    return n.toString().split('.')[1].length || 0;
+  }
+
+  /**
+   * https://stackoverflow.com/questions/11832914/round-to-at-most-2-decimal-places-only-if-necessary
+   * @param n {number}
+   * @param decimalPlaces {number}
+   * @returns {number}
+   */
+  function roundDecimals(n, decimalPlaces) {
+    var diff = Math.pow(10, decimalPlaces);
+    return Math.round((n.valueOf() + Number.EPSILON) * diff) / diff;
+  }
+
+  /**
+   * Generic handler that sets 'is-invalid' class to input, if value is NaN or <0
+   * @param onInputValue
+   * @returns {function(value : Number | NaN): (undefined)}
+   */
+  function inputEventHandler(onInputValue) {
+    return function(e) {
+      var inputEl = $(this);
+      inputEl.removeClass('is-invalid');
+
+      var numberValue = Number(e.target.value);
+      if (isNaN(numberValue) || numberValue < 0
+        || countDecimals(numberValue) > 2) {
+        inputEl.addClass('is-invalid');
+        onInputValue(NaN);
+        return;
+      }
+
+      onInputValue(numberValue);
+    };
+  }
+
+  (function initAv1Form() {
+    // form values for calculation
+    let encodingCostPerMinute = 0.02;
+    let ingressCostPerGb = 0.00;
+    let egressCostPerGb = 0.04;
+
+    let numberOfStreamsUhd = 2;
+    let numberOfStreamsHd = 3;
+    let numberOfStreamsSd = 4;
+
+    const multiplierStreamUhd = 4;
+    const multiplierStreamHd = 2;
+    const multiplierStreamSd = 1;
+    const multiplierTechPerTitle = 1.1;
+    const multiplierTech3Pass = 2; // Multipass
+    const multiplierCodecAv1 = 10;
```

Comment:
does this already yield the correct results?
i'm missing the H.265 codec multiplier

Reply:
I added the multipliers in the code, but then removed all unused values after implementing the calculation recursively from the spreadsheet.